projects/RabbitMQ.Client.OpenTelemetry/TraceProviderBuilderExtensions.cs

```diff
@@ -0,0 +1,85 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using OpenTelemetry.Context.Propagation;
+using RabbitMQ.Client;
+using RabbitMQ.Client.OpenTelemetry;
+
+namespace OpenTelemetry.Trace
+{
+    public static class OpenTelemetryExtensions
+    {
+        internal static TextMapPropagator s_propagator = Propagators.DefaultTextMapPropagator;
+
+        public static TracerProviderBuilder AddRabbitMQ(this TracerProviderBuilder builder,
+            RabbitMQOpenTelemetryConfiguration configuration)
+        {
+            if (configuration.PropagateBaggage)
+            {
+                s_propagator = new CompositeTextMapPropagator(new TextMapPropagator[]
+                {
+                    new TraceContextPropagator(), new BaggagePropagator()
+                });
+            }
+            else
+            {
+                s_propagator = new TraceContextPropagator();
+            }
+
+            RabbitMQActivitySource.UseRoutingKeyAsOperationName = configuration.UseRoutingKeyAsOperationName;
```

Comment:
do you feel it's something we should document better /provide more clear guidance on in the messaging semconv?

Reply:
Might be a good idea if people are running into high-cardinality issues with their messaging span names.

Reply:
Is Operation the right name here? Since this is focused on OpenTelemetry, I'm wondering if just "SpanName" is a better wording.

Reply:
OTel config: `UseRoutingKeyInSpanName`
RabbitMQActivitySource property: `UseRoutingKeyInActivityName`
Make more sense?

Reply:
> Might be a good idea if people are running into high-cardinality issues with their messaging span names.

the span name cardinality requirements are soft (~undefined~) - https://github.com/open-telemetry/opentelemetry-specification/issues/3534. Things like HTTP route we put there are like middlish-cardinality and I think the same is true about queue names/routing key.

so if cardinality is the only problem, I'd not make span name configurable (worst case users can rename spans with a processor). So again, no strong opinions